api/src/main/java/org/pmiops/workbench/actionaudit/bucket/BucketAuditQueryServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package org.pmiops.workbench.actionaudit.bucket;
+
+import static org.pmiops.workbench.exfiltration.ObjectNameSizeConstants.THRESHOLD;
+
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.TableResult;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchConfig.BucketAuditConfig;
+import org.pmiops.workbench.utils.FieldValues;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BucketAuditQueryServiceImpl implements BucketAuditQueryService {
+
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private final BigQueryService bigQueryService;
+
```

Comment:
The query here groups by the pet account, project ID and bucket and sums the file lengths of the modified resources. It only returns the entries with sum > THRESHOLD